core/utils/timestamp.js

```diff
@@ -60,3 +61,14 @@ function maxDate(d1 /*: Date */, d2 /*: Date */) /*: Date */ {
 function stringify(t /*: Timestamp */) {
   return t.toISOString().substring(0, 19) + 'Z'
 }
+
+function roundedRemoteDate(isoDate /*: string */) /*: string */ {
+  const date = new Date(isoDate)
+
+  const millisecond = isoDate.split(/\.|Z/)[1]
+  if (millisecond.length > 3) {
+    return new Date(date.getTime() + 1).toISOString()
```

Comment:
In this situation (i.e. more than 3 milliseconds digits) we need to "truncate" the date to fit our 3 milliseconds digits standard.
However, simply truncating is not an option because `cozy-stack` does not allow changing a Date attribute with an older value and truncating would mean, in most cases, getting a lower value (with the very rare exception of a 0 milliseconds date).

So we make sure we have a greater value that's close enough by adding 1 millisecond to the rounded date (hence the "rounded" part in the function name).

I can add a comment if you feel that's necessary.

Reply:
Alright, yeah I definitely think a comment is need there 😁